### PR TITLE
Merging lists and updating on CLI command.

### LINKED ISSUE #7412: Merge config-file labels and kubernetes into the RRDHOST.
* Merge the two linked lists, so that user-defined labels take precedence over the kubernetes labels.
* Acquire the lock in RRDHOST.
* Replace the old list.
* Release the lock and clean up the old list.

### PR DESCRIPTION
**Pull request by amoss**
##### Summary
Fix #7412

### CODE REVIEW

File: database/rrd.h [hunk 1]
```diff
@@ -164,6 +164,8 @@ struct label {
 };
 
 struct label *create_label(char *key, char *value, LABEL_SOURCE label_source);
+struct label *add_label_to_list(struct label *l, char *key, char *value, LABEL_SOURCE label_source);
+void reload_host_labels();
```
Comment (thiagoftsm) [inline database/rrd.h]: Are you not missing an `extern` before this function declaration?

File: database/rrdhost.c [hunk 2]
```diff
@@ -741,6 +756,66 @@ struct label *create_label(char *key, char *value, LABEL_SOURCE label_source)
...
+struct label *add_label_to_list(struct label *l, char *key, char *value, LABEL_SOURCE label_source)
+{
+    struct label *lab = create_label(key, value, label_source);
+    lab->next = l;
+    return lab;
+}
+
+int list_contains(struct label *head, struct label *check)
+{
+    while (head!=NULL)
```
Comment (thiagoftsm) [inline database/rrdhost.c]: Please add a space between values and comparison.

File: database/rrdhost.c [hunk 3]
```diff
@@ -741,6 +756,66 @@ struct label *create_label(char *key, char *value, LABEL_SOURCE label_source)
...
 }
 
+struct label *add_label_to_list(struct label *l, char *key, char *value, LABEL_SOURCE label_source)
+{
+    struct label *lab = create_label(key, value, label_source);
+    lab->next = l;
+    return lab;
+}
+
+int list_contains(struct label *head, struct label *check)
```
Comment (thiagoftsm) [inline database/rrdhost.c]: This only a suggestion, Netdata works with different link lists and the name of this function does not say which list we are comparing, but the arguments are clear. My suggestion is to add the work `label` to the name of this function for its name to be more meaningful.

File: database/rrd.h (hunk 1 above)
Comment (cakrit) [reply database/rrd.h]: I'll try it and change it if necessary, I expect he's tested it.

File: database/rrdhost.c (hunk 3 above)
Comment (cakrit) [reply database/rrdhost.c]: Will rename this.

File: database/rrdhost.c (hunk 2 above)
Comment (cakrit) [reply database/rrdhost.c]: Will fix.

File: database/rrd.h (hunk 1 above)
Comment (amoss) [reply database/rrd.h]: The extern keyword is not needed. Yes I tested it :)

File: daemon/commands.c [hunk 4]
```diff
@@ -45,6 +46,7 @@ static command_info_t command_info_array[] = {
         {"reopen-logs", cmd_reopen_logs_execute, CMD_TYPE_ORTHOGONAL},       // Close and reopen log files
         {"shutdown-agent", cmd_exit_execute, CMD_TYPE_EXCLUSIVE},            // exit cleanly
         {"fatal-agent", cmd_fatal_execute, CMD_TYPE_HIGH_PRIORITY},          // exit with fatal error
+        {"reload-labels", cmd_reload_labels_execute, CMD_TYPE_ORTHOGONAL},   // reload the labels
```
Comment (mfundul) [inline daemon/commands.c]: You should add this description in the `help` command output and `README.md`.

**Pull request merged**